### ISSUE TITLE
word layouts not deployed with .app properly?

### ISSUE THREAD

**Issue by CanuckNav74** [Microsoft/AL#4204]
I'm currently testing out an extension I wrote. It contains one word layout report. I originally developed this report in cside and ran through the process to export using the new syntax and then running it through the txt2al tool. 

The extension compiled fine, the .docx word layout exported to the filesystem is fine. 

However, when I ran the report in the web client, I was greeted with an empty page.

Sure enough, when I exported the layout from the web client, the downloaded .docx was empty. 

When I uploaded a custom layout (i.e. the good .docx file in VS code), the report was no longer blank and thus worked. This however is only a workaround and I don't see this as a satisfactory solution. 

In my vscode folder for the project, the .al and .docx are in the following path:

reports/al/Rep50000.ATL Standard Sales - Invoice/*

In the report .al source code, the relative path to the word layout file is also correct:

WordLayout = './ATL Standard Sales - Invoice.docx';

interestingly enough, I opened the compiled .app file in 7zip to see where these files went in the app file, the file paths are nothing like the directory structure in my AL project.

the .al file is in \src\reports\Rep50000.ATL%2520Standard%2520Sales%2520-%2520Invoice\al\
the layout is in \layout\

Why is business central ignoring my layout when the deploy my extension from VS Code? Did I just uncover a bug? I'm guessing the compiler built an app where it wouldn't know where to locate the word layout (explaining the blank layout). 

**Versions:**
 - AL Langauge: 2.0.48254
 - Business Central: CA Dynamics NAV 13.0 (24844) (local sandbox docker container)

**Comment by CanuckNav74** [comment]
I just wanted to leave an added comment. When the .app was initially built, it created a copy of the .docx file and put it into the root dir of my AL project. Sure enough, that one was the blank template. 

Seems that The AL Language extension doesn't like my particulat folder structure. To resolve this I had to overwite the blank template file with a good one and recompile. 

Question still stands - is this a bug?

**Comment by StanislawStempin** [comment]
That's because the layout path is relative to the project root and not to the report object. 

You can see where AL thinks the layout should be by observing where the empty template is created. This is where the actual layout should be.

We have a change request to make the path relative to the report object but need to make it in a way that doesn't break backwards compatibility with the current behavior.

**Comment by CanuckNav74** [comment]
Thankyou for the clarification.

Microsoft's choice of ./ path notation for layout paths is very misleading if that's the case. I would even say incorrect

In my opinion, ./layout.docx should be relative to the .Al file, and ~/layout.docx should be relative to the project root.

**Comment by JavierFuentes** [comment]
Regarding to this issue I reported in 2018...

https://github.com/microsoft/AL/issues/4225

I just tested it again with 3 files inside a subfolder of muy project:
./Reports/MyReport.al
./Reports/MyReport.docx
./Reports/MyReport.rdlc

I've deployed my extension into a last version v19.  docker container.

When I download my sources, the .ZIP file has MyReport.al at root folder, but keeps the other files at ./Reports path

Definitely, the sources download .ZIP file does not fully respect the original folder structure of the project, and therefore, the file loss problem has not been fully resolved.
